### PR TITLE
Peak performance paper

### PR DESCRIPTION
**Pull request by Y0dler**
Add the PeakPerformance paper and associated files to the repository in order to submit it to JOSS.

### CODE REVIEW

File: paper/paper.md [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+If the filter is not passed a second time, the pertaining signal is rejected.
+Harnessing the advantages of the uncertainty quantification, a second criterion calculates the ratio of the resulting standard deviation of a peak parameter to its mean and discards signals exceeding a threshold.
+Usually, false positives passing the first criterion are rather noisy signals where a fit was achieved but the uncertainty on the peak parameters is extremely high.
+These signals will then be rejected by the second criterion, ultimately reducing the number of false positive peaks significantly if not eliminating them.
+If a signal was accepted as a peak, the final simulation step is a posterior predictive check which is added to the inference data object resulting from the model simulation.
+
+
+## Peak fitting results and diagnostic plots
+After completing a cycle of the data pipeline or prematurely exiting it through one of the filters, the results need to be communicated and made available to the user.
+This is done in multiple ways.
```
Comment (michaelosthege) [inline paper/paper.md:264]:
```suggestion
This is done in multiple ways:
```

File: paper/paper.md [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+Additional columns provide information on the acquisition and mass trace in question.
+Finally, there are columns stating whether the signal was recognized as a peak, if applicable the reason for the rejection of the signal, the utilized model for the simulation, and in case of a double peak a column specifying the peak number ("1st" or "2nd").
+Accordingly, when a signal is rejected, it will nonetheless be added to the Excel report file and the exact reason for its rejection is detailed.
+The second Excel file created is denominated as "area_summary.xlsx" and is a more handy version of "peak_data_summary.xlsx" with a reduced degree of detail.
+As implied by the name, from the peak parameters only the peak area remains and the columns are trimmed down to the essentials.
+Since subsequent data analyses will most likely rely on the peak area, first and foremost, this sheet should facilitate the further usage of the data.
+The most valuable result, however, is the storage of the inference data objects created only when a signal was identified as a peak.
```
Comment (michaelosthege) [inline paper/paper.md:273]:
```suggestion
The most valuable result, however, are the inference data objects saved to disk for each signal for which a peak function was successfully fitted.
```

File: paper/paper.md [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+Conveniently, the inference data objects saved as $\texttt{*.nc}$ files contain all data and metadata related to the process enabling the user to be able to perform diagnostics or create custom plots not featured in $\texttt{PeakPerformance}$.
```
Comment (michaelosthege) [inline paper/paper.md:274]:
```suggestion
Conveniently, the inference data objects saved as $\texttt{*.nc}$ files contain all data and metadata related to the Bayesian parameter estimation, enabling the user to perform diagnostics or create custom visualizations not already provided by $\texttt{PeakPerformance}$.
```

File: paper/paper.md [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+Regarding data visualization with the matplotlib package [@matplotlib, @matplotlibzenodo], $\texttt{PeakPerformance}$'s $\texttt{plots}$ module offers the generation of two diagram types for each successfully fitted peak.
+The posterior plot presents the fit of the intensity function alongside the raw data points.
+The first row of Figure 4 presents two such examples where the single peak diagram shows the histidine (His) fragment with a m/z ratio of 110 Da and the double peak diagram the leucine (Leu) and isoleucine (Ile) fragments with a m/z ratio of 86 Da.
+
+![](./Fig4_peak_results.png)
+__Figure 4:__ Results plots for a single His peak and a double Leu and Ile peak depicting the peak fit (first row) and the posterior predictive checks (second row) alongside the raw data. The numerical results are listed in table 2.
```
Comment (michaelosthege) [inline paper/paper.md:280]: stimmen die 95 % ?

```suggestion
__Figure 4:__ Results plots for a single His peak and a double Leu and Ile peak depicting the peak fit (first row) and the posterior predictive checks (second row) alongside the raw data. The numerical results are listed in table 2. The dashed lines mark the 95 \% highest density interval of the posterior (red) and posterior predictive (blue) distributions.
```

File: paper/paper.md [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+
+The posterior predictive plots in the second row of Figure 4, then, exhibit the posterior predictive checks alongside the raw data.
```
Comment (michaelosthege) [inline paper/paper.md:282]:
```suggestion
The posterior predictive plots in the second row of Figure 4 are provided for visual posterior predictive checks, namely the comparison of observed and predicted data distribution.
```

File: paper/paper.md [hunk 6]
```diff
@@ -0,0 +1,368 @@
...
+
+Another important feature of $\texttt{PeakPerformance}$ is constituted by the easy access to diagnostic metrics for extensive quality control.
+Using the data stored in an inference data object of a fit, the user can utilize the ArviZ package to generate various diagnostic plots.
+One particularly useful one is the cumulative posterior predictive plot portrayed in Figure 5.
+This plot enables users to judge the quality of a fit and identify instances of lack-of-fit.
+As can be seen in the left plot, some predicted intensity values in the lowest quantile of the single peak example show a minimal lack-of-fit.
+Importantly, such a deviation can be observed, judged and is quantifiable which intrinsically represents a large improvement over the status quo.
+
+![](./Fig5_ppc.png)
+__Figure 5:__ Cumulative posterior predictive plots created with the ArviZ package and pertaining to the example data of the single His peak (left) and the double Leu and Ile peak (right).
```
Comment (michaelosthege) [inline paper/paper.md:301]:
```suggestion
__Figure 5:__ Cumulative posterior predictive plots created with the ArviZ package and pertaining to the example data of the single His peak (left) and the double Leu and Ile peak (right). The empirical cumulative density function (black) is in good agreement with the median posterior predictive (orange) and lies within the predicted variance (blue band), visualizing that the model provides an adequate prediction irrespective of the intensity value.
```

File: paper/paper.md [hunk 7]
```diff
@@ -0,0 +1,368 @@
...
+
+In the second stage, marginal cases in the form of slightly skewed peaks were investigated to observe whether their estimation with a normal- or skew normal-shaped intensity function would result in significant differences in terms of peak area and height.
+Here, a slight skew was defined as an $\alpha$ parameter of 1 resulting in peak shapes not visibly discernible as clearly normal or skew normal.
+With a sample size of 100 noisy, randomly generated data sets, we show that nearly identical estimates for peak area and height, as well as their respective uncertainties are obtained regardless of the utilized model (Fig. 6b).
+The exhibited mean values are based on fractions of the key peak parameters area and height between results obtained with a normal and skew normal model which were defined as
+
+$$\tag{13}F_{n / \mathrm{sn}} = \frac{A_{\mathrm{normal}}}{A_{\mathrm{skew \ normal}}}$$
```
Comment (michaelosthege) [inline paper/paper.md:323]:
```suggestion
$$\tag{13}F_{n / \mathrm{sn}} = \frac{A_{\mathcal{N}}}{A_{\mathrm{skew \ normal}}}$$
```

File: paper/paper.md [hunk 8]
```diff
@@ -0,0 +1,368 @@
...
+By showing not only the mean area ratio of all peaks but also the ones for the single and double peak subgroups, it is evident that the variance is significantly higher for double peaks.
+In case of this data set, two low quality double peaks in particular inflated the variance significantly which may not be representative for other data sets.
+It has to be stated, too, that the prevalence of manual re-integration of double peaks in MQ might have introduced a user-specific bias, thereby increasing the final variance.
+Nevertheless, it could be shown that $\texttt{PeakPerformance}$ yields comparable peak area results to a commercially available vendor software.
+
+# Conclusions
+$\texttt{PeakPerformance}$ is a tool for automated LC-MS/MS peak data analysis employing Bayesian inference.
+Due to its Bayesian methodology, it provides built-in uncertainty quantification and thus for the first time takes the measurement noise of an LC-MS/MS device into account when integrating peaks.
```
Comment (michaelosthege) [inline paper/paper.md:344]:
```suggestion
It provides built-in uncertainty quantification by Bayesian parameter estimation and thus for the first time takes the measurement noise of an LC-MS/MS device into account when integrating peaks.
```

File: paper/summary_joint.png
Comment (michaelosthege) [inline paper/summary_joint.png:1]: Hier steht in den Tabellenüberschriften noch "distribution" → lieber "model" oder "model function" schreiben

**Review by michaelosthege — COMMENTED** [review]
👍👍 with a few smaller edits

What is the raw format for that table? Inkscape SVG?
If we can print it to a PDF that would render the text selectable for the output.

File: paper/paper.md (hunk 4 above)
Comment (Y0dler) [reply paper/paper.md:280]: Mittlerweile würden sie das, aber damals waren es ziemlich sicher 94 %. Und die Posterior-Plots haben keine dashed lines, die sind nur bei den ppc checks, wenn ich das richtig sehe.

File: paper/paper.md [hunk 9]
```diff
@@ -0,0 +1,368 @@
...
+  - Bayesian inference
+  - Chromatography
+  - LC-MS/MS
+  - Python
+  - Uncertainty quantification
+
+authors:
+  - name: Jochen Nießer
+    orcid: 0000-0001-5397-0682
+    affiliation: 1
```
[the remark this answers is not in the document]
Comment (Y0dler) [reply paper/paper.md:14]: Guter Punkt.

File: paper/summary_joint.png
Comment (Y0dler) [reply paper/summary_joint.png:1]: Ändere ich. Im Original ist die Tabelle ein Excelsheet, also kann ich sie dann auch als PDF exportieren (in Bezug auf deinen obigen Kommentar).

**Comment by michaelosthege** [comment]
I tried to format Table 2 as a pandoc grid table.

Using VS Code multiline cursor this was reasonably comfortable to edit.

I used https://pandoc.org/try/ to convert the table to HTML and viewed it using https://html.onlineviewer.net/ and got something that might actually compile 👀

**Comment by michaelosthege** [comment]
With the following raw markup this is the best I got:

```
+--------------------+-------------+-----------+----------+----------+-------------+------------+----------+----------+
| parameter          | skew normal model                             | double normal model                            |
|                    +-------------+-----------+----------+----------+-------------+------------+----------+----------+
|                    | mean        | sd        | 3 %      | 97 %     | mean        | sd         | 3 %      | 97 %     |
+:===================+============:+==========:+=========:+=========:+============:+===========:+=========:+=========:+
| intercept          | -43.94      | 7.41      | -57.88   | -30.02   | 1115.40     | 38.69      | 1040.14  | 1185.07  |
+--------------------+-------------+-----------+----------+----------+-------------+------------+----------+----------+
| slope              | 6.66        | 0.51      | 5.71     | 7.63     | -21.65      | 3.09       | -27.50   |-15.94    |
+--------------------+-------------+-----------+----------+----------+-------------+------------+----------+----------+
| noise              | 103.63      | 7.51      | 89.50    | 117.26   |  118.63     | 8.01       | 103.52   | 133.29   |
+--------------------+-------------+-----------+----------+----------+-------------+------------+----------+----------+
|                    |             |           |          |          |  __11.73__  | __0.02__   | 11.70    | 11.76    |
| mean               | __25.95__   | __0.01__  | 25.93    | 25.97    +-------------+------------+----------+----------+
|                    |             |           |          |          |  __12.43__  | __0.01__   | 12.42    | 12.45    |
+--------------------+-------------+-----------+----------+----------+-------------+------------+----------+----------+
|                    |             |           |          |          |  __317.16__ | __28.84__  | 263.23   | 370.56   |
| area               | __1512.32__ | __37.31__ | 1141.25  | 1581.37  +-------------+------------+----------+----------+
|                    |             |           |          |          |  __674.34__ | __26.34__  | 623.47   | 722.88   |
+--------------------+-------------+-----------+----------+----------+-------------+------------+----------+----------+
|                    |             |           |          |          |  __774.99__ | __65.28__  | 653.50   | 897.88   |
| height             | __1879.72__ | __37.71__ | 1809.30  | 1950.64  +-------------+------------+----------+----------+
|                    |             |           |          |          |  __1762.66__| __64.04__  | 1639.62  | 1881.66  |
+--------------------+-------------+-----------+----------+----------+-------------+------------+----------+----------+
|                    |             |           |          |          |  0.16       | 0.02       | 0.13     | 0.20     |
| std                | 0.53        | 0.02      | 0.48     | 0.56     +-------------+------------+----------+----------+
|                    |             |           |          |          |  0.15       | 0.01       | 0.14     | 0.17     |
+--------------------+-------------+-----------+----------+----------+-------------+------------+----------+----------+
|                    |             |           |          |          |  6.56       | 0.72       | 5.22     | 7.88     |
| sn                 | 18.24       | 1.37      | 15.69    | 20.76    +-------------+------------+----------+----------+
|                    |             |           |          |          |  14.93      | 1.14       | 12.82    | 17.14    |
+--------------------+-------------+-----------+----------+----------+-------------+------------+----------+----------+
| alpha              | 2.96        | 0.39      | 2.27     | 3.71     |  -          | -          | -        | -        |
+--------------------+-------------+-----------+----------+----------+-------------+------------+----------+----------+
```

grafik


I reverted to embedding Table 2 from the SVG...

**Comment by michaelosthege** [comment]
Documentation build :x: because Sphinx released 8.1.0 which broke `sphinxcontrib.mermaid 0.9.2` (https://github.com/mermaid-js/mermaid/issues/5954) the `sphinxcontrib.mermaid 1.0.0` release is almost there (https://github.com/mgaitan/sphinxcontrib-mermaid/pull/153) and will fix this, so let's just wait.

**Review by michaelosthege — APPROVED** [review]
We discussed to merge this and continue refactoring the paper/documentation on `main`.

**Pull request merged**